core/store/models/ocrkey/key_bundle.go

```diff
@@ -0,0 +1,252 @@
+package ocrkey
+
+import (
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/big"
+	"time"
+
+	cryptorand "crypto/rand"
+
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/crypto/secp256k1"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/curve25519"
+)
+
+// KeyBundle represents the bundle of keys needed for OCR
+type KeyBundle struct {
+	ID                 string
+	onChainSigning     *onChainPrivateKey
+	offChainSigning    *offChainPrivateKey
+	offChainEncryption *[curve25519.ScalarSize]byte
+}
+
+// EncryptedKeyBundle holds an encrypted KeyBundle
+type EncryptedKeyBundle struct {
+	ID                    string `gorm:"primary_key"`
+	OnChainSigningAddress OnChainSigningAddress
+	OffChainPublicKey     OffChainPublicKey
+	EncryptedPrivateKeys  []byte
+	CreatedAt             time.Time
+	UpdatedAt             time.Time
+}
+
+type keyBundleRawData struct {
+	EcdsaX             big.Int
+	EcdsaY             big.Int
+	EcdsaD             big.Int
+	Ed25519PrivKey     []byte
+	OffChainEncryption [curve25519.ScalarSize]byte
+}
+
+type scryptParams struct{ N, P int }
+
+var defaultScryptParams = scryptParams{
+	N: keystore.StandardScryptN, P: keystore.StandardScryptP}
+
+var curve = secp256k1.S256()
+
+func (EncryptedKeyBundle) TableName() string {
+	return "encrypted_ocr_key_bundles"
+}
+
+// NewKeyBundle makes a new set of OCR key bundles from cryptographically secure entropy
+func NewKeyBundle() (*KeyBundle, error) {
+	reader := cryptorand.Reader
+	onChainSk, err := cryptorand.Int(reader, crypto.S256().Params().N)
```

Comment:
Have you considered using something like 

```golang
k, err := ecdsa.GenerateKey(curve, reader)
if err != nil ...
onChainPriv := (*onChainPrivateKey)(k)
```

?

https://golang.org/pkg/crypto/ecdsa/#GenerateKey

Reply:
great call - so much cleaner